src/Dependencies/Collections/Internal/SegmentedArrayHelper.cs

```diff
@@ -94,7 +112,7 @@ private static int CalculateSegmentSize(int elementSize)
             static int ArraySize(int elementSize, int segmentSize)
             {
                 // Array object header, plus space for the elements
-                return (2 * IntPtr.Size) + (elementSize * segmentSize);
+                return (2 * IntPtr.Size + 8) + (elementSize * segmentSize);
```

Comment:
> + 8 [](http://example.com/codeflow?start=39&length=4)

It is not obvious why this change is necessary.

Reply:
➡️ Updated the PR commit to mention that this is a follow-up to #67558, where comment https://github.com/dotnet/roslyn/pull/67558#issuecomment-1489441636 called out the incorrect formula